language-plutus-core/src/Language/PlutusCore/MkPlc.hs

```diff
@@ -126,3 +136,10 @@ mkIterTyLam
     -> Type tyname a
     -> Type tyname a
 mkIterTyLam x args body = foldr (\(TyVarDecl _ n k) acc -> TyLam x n k acc) body args
+
+mkIterKindArrow
+    :: a
+    -> [Kind a]
+    -> Kind a
+    -> Kind a
+mkIterKindArrow x kinds target = foldr (\kind acc -> KindArrow x kind acc) target kinds
```

Comment:
I think this can just use `foldr (KindArrow x) target kinds`

Reply:
I think when I wrote this, there was a local convention in this module that prescribed to name arguments. Ok, I'll eta-contract.